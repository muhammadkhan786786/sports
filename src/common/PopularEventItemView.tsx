import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {SportEventTypes} from '../types';

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: 'rgb(22,22,22)',
    marginHorizontal: 8,
    marginVertical: 4,
    justifyContent: 'center',
    paddingVertical: 16,
    paddingHorizontal: 12,
    borderRadius: 4,
  },
  headingAndStateParent: {
    flex: 1,
    flexDirection: 'row',
  },
  headingParent: {
    flex: 1,
  },
  headingText: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
  subHeading: {
    color: 'white',
    fontStyle: 'italic',
  },
  description: {
    color: 'white',
    marginTop: 4,
  },
});

interface Props {
  item: SportEventTypes;
  navigation?: any;
}

export default function PopularEventItemView({
  item,
  navigation,
}: Props): JSX.Element {
  return (
    <TouchableOpacity
      onPress={() => navigation.navigate('SportEventDetails', {data: item})}>
      <View style={styles.root}>
        <View style={styles.headingAndStateParent}>
          <View style={styles.headingParent}>
            <Text style={styles.headingText}>{item.name}</Text>
          </View>
          <View>
            <Text style={styles.headingText}>{item.state}</Text>
          </View>
        </View>

        <Text style={styles.subHeading}>{item.short_name}</Text>
        <Text style={styles.description}>
          {item.description
            ? item.description
            : 'Description of the event is not provided at the moment.'}
        </Text>
      </View>
    </TouchableOpacity>
  );
}
