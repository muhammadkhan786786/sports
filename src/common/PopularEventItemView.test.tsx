import React from 'react';
import renderer from 'react-test-renderer';
import PopularEventItemView from './PopularEventItemView';

const data = {
  bet_allowed: true,
  bettable: true,
  chart_time_period: null,
  created: '2021-05-10T00:34:50.887745Z',
  description: null,
  display_order: 0,
  end_date: null,
  full_slug:
    '/sport/football/germany-bundesliga/2021/05/16/18-30/rb-leipzig-vs-vfl-wolfsburg',
  hidden: false,
  id: '42207308',
  inplay_enabled: true,
  modified: '2021-05-16T20:23:44.059361Z',
  name: 'RB Leipzig vs VfL Wolfsburg',
  parent_id: '30088540',
  seo_description: null,
  short_name: 'RBL vs. VFL',
  slug: 'rb-leipzig-vs-vfl-wolfsburg',
  special_rules: null,
  start_date: '2021-05-16',
  start_datetime: '2021-05-16T18:30:00Z',
  state: 'ended',
  type: 'football_match',
};

describe('PopularEventItemView component', () => {
  test('PopularEventItemView should renders all the views if item prop is given', () => {
    const tree = renderer.create(<PopularEventItemView item={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
