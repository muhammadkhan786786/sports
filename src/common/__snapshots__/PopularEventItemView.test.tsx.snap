// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularEventItemView component PopularEventItemView should renders all the views if item prop is given 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(22,22,22)",
        "borderRadius": 4,
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 8,
        "marginVertical": 4,
        "paddingHorizontal": 12,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          RB Leipzig vs VfL Wolfsburg
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          ended
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontStyle": "italic",
        }
      }
    >
      RBL vs. VFL
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "marginTop": 4,
        }
      }
    >
      Description of the event is not provided at the moment.
    </Text>
  </View>
</View>
`;
